wee_slack.py

```diff
@@ -2957,27 +2987,45 @@ def msg_command_cb(data, current_buffer, args):
 @utf8_decode
 def command_talk(data, current_buffer, args):
     """
-    Open a chat with the specified user
-    /slack talk [user]
+    Open a chat with the specified user(s)
+    /slack talk <user>[,<user2>[,<user3>...]]
     """
 
     e = EVENTROUTER
     team = e.weechat_controller.buffers[current_buffer].team
     channel_name = args.split(' ')[1]
-    c = team.get_channel_map()
-    if channel_name not in c:
-        u = team.get_username_map()
-        if channel_name in u:
-            s = SlackRequest(team.token, "im.open", {"user": u[channel_name]}, team_hash=team.team_hash)
-            EVENTROUTER.receive(s)
-            dbg("found user")
-            # refresh channel map here
-            c = team.get_channel_map()
 
     if channel_name.startswith('#'):
         channel_name = channel_name[1:]
-    if channel_name in c:
-        chan = team.channels[c[channel_name]]
+
+    chan = team.find_channel_by_name(channel_name)
+
+    # If the channel name doesn't exist, try finding a DM or MPDM instead
+    if not chan:
+        # Get the IDs of the users
+        u = team.get_username_map()
+        users = {u[user] for user in channel_name.split(',') if user in u}
+
+        if users:
+            if len(users) > 1:
+                channel_type = 'mpim'
+                # Add the current user if not given since they'll be in MPDMs
+                if team.myidentifier not in users:
+                    users.add(team.myidentifier)
+            else:
+                channel_type = 'im'
+
+            chan = team.find_channel_by_members(channel_type, users)
+
+            # If the DM or MPDM doesn't exist, create it
+            if not chan:
+                s = SlackRequest(team.token, SLACK_API_TRANSLATOR[channel_type]['join'], {'users': ','.join(users)}, team_hash=team.team_hash)
+                EVENTROUTER.receive(s)
+                dbg("found user")
```

Comment:
I see that this message was here before as well, but I can't really see when it is useful.

Reply:
Me neither, will also remove.